nixos/modules/services/web-apps/zabbix.nix

```
@@ -154,69 +250,104 @@ in
           Additional configuration to be copied verbatim into {file}`zabbix.conf.php`.
         '';
       };
-
     };
   };
 
   # implementation
 
   config = mkIf cfg.enable {
 
-    services.zabbixWeb.extraConfig = optionalString ((versionAtLeast config.system.stateVersion "20.09") && (versionAtLeast cfg.package.version "5.0.0")) ''
-      $DB['DOUBLE_IEEE754'] = 'true';
-    '';
+    services.zabbixWeb.extraConfig =
+      optionalString
+        (
+          (versionAtLeast config.system.stateVersion "20.09") && (versionAtLeast cfg.package.version "5.0.0")
+        )
+        ''
+          $DB['DOUBLE_IEEE754'] = 'true';
+        '';
 
-    systemd.tmpfiles.rules = [
-      "d '${stateDir}' 0750 ${user} ${group} - -"
-      "d '${stateDir}/session' 0750 ${user} ${config.services.httpd.group} - -"
-    ];
+    systemd.tmpfiles.rules =
+      [ "d '${stateDir}' 0750 ${user} ${group} - -" ]
+      ++ optionals (cfg.frontend == "httpd") [
+        "d '${stateDir}/session' 0750 ${user} ${config.services.httpd.group} - -"
+      ]
+      ++ optionals (cfg.frontend == "nginx") [
+        "d '${stateDir}/session' 0750 ${user} ${config.services.nginx.group} - -"
+      ];
 
     services.phpfpm.pools.zabbix = {
       inherit user;
-      group = config.services.httpd.group;
-      phpOptions = ''
-        # https://www.zabbix.com/documentation/current/manual/installation/install
-        memory_limit = 128M
-        post_max_size = 16M
-        upload_max_filesize = 2M
-        max_execution_time = 300
-        max_input_time = 300
-        session.auto_start = 0
-        mbstring.func_overload = 0
-        always_populate_raw_post_data = -1
-        # https://bbs.archlinux.org/viewtopic.php?pid=1745214#p1745214
-        session.save_path = ${stateDir}/session
-      '' + optionalString (config.time.timeZone != null) ''
-        date.timezone = "${config.time.timeZone}"
-      '' + optionalString (cfg.database.type == "oracle") ''
-        extension=${pkgs.phpPackages.oci8}/lib/php/extensions/oci8.so
-      '';
+      group =
+        if cfg.frontend == "httpd" then config.services.httpd.group else config.services.nginx.group;
```

```suggestion
      group = config.services.${cfg.frontend}.group;
```

what do you think about that? personally i prefer to see that instead of a bunch of 'if's... please take the suggestion if you like it, otherwise leave it

---

LGTM!